### ISSUE TITLE
consider-using-f-string produces worse quality code in some cases

### ISSUE THREAD

**Issue by doublethefish** [PyCQA/pylint#5144]
### Bug description

`consider-using-f-string` produces worse quality code in some cases, for example consider the following format string:
```python

            "Timing of '{name}' took {secs}s/{total_secs}s (started {start_time})".format(
                name=name,
                secs=delta.total_seconds(),
                total_secs=self._config.timeout_seconds,
                start_time=self._start_time.strftime("%H:%M:%S %Z").strip(),
            )
```

Any variant of using an f-string to remove the `consider-using-f-string` warning does one of the following:
* causes more local variables to be created (possibly blowing the max-locals limit)
* is less readable (as the context of what the pieces are intended for is lost, e.g. `took {secs}s/{total_secs}s` is significantly more readable than `took {delta.total_seconds()}s/{self._config.timeout_seconds}s`
* splits the context over multiple lines, making it harder to read the context

### Command used

```shell
python3 -m pylint -j4 --score=n --rcfile=.pylintrc <files>
```

### Pylint output

```shell
[..snip..]
C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
[..snip..]
```

### Expected behavior

Only emit the warning for invocations of [str.format](https://docs.python.org/3/library/stdtypes.html#str.format) that are below a certain complexity i.e. 5 parameters.

Additionally only emit the warning when the conformed string is 
1. within line-width limits (including indents), or
2. roughly the same length as the string it would replace - say maximally 10% larger than the existing string (as f-strings are smaller in most cases).

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.2
Python 3.10.0 (default, Oct  6 2021, 00:30:31) [GCC 10.3.1 20210424]
```

### OS / Environment

n/a but
Alpine Linux 9b0fb0d01a22 5.10.47-linuxkit #1 SMP PREEMPT Sat Jul 3 21:50:16 UTC 2021 aarch64 Linux
bash

**Comment by Pierre-Sassoulas** [comment]
Thank you for searching for other issue related to that @doublethefish 😄. For reference if someone happen to see this issue following a search, in the example given it's possible to do:
```python
            template = "Timing of '{name}' took {secs}s/{total_secs}s (started {start_time})"
            template.format(
                name=name,
                secs=delta.total_seconds(),
                total_secs=self._config.timeout_seconds,
                start_time=self._start_time.strftime("%H:%M:%S %Z").strip(),
            )
```
And not have any warnings raised.
